hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,244 @@
+        Raises: SwitchError
+        Returns: If successful returns None else logs error message
+        """
+        try:
+            for command in command_strings:
+                arg_list = shlex.split(command)
+                subprocess.check_call(arg_list)
+        except subprocess.CalledProcessError as e:
+            logger.error('%s', e)
+            raise SwitchError('ovs command failed: ')
+
+    def get_port_networks(self, ports):
+
+        response = {}
+        for port in ports:
+            trunk_id_list = self._interface_info(port.label)['trunks']
+            if trunk_id_list == []:
+                response[port] = []
+            else:
+                result = []
+                for trunk in trunk_id_list:
+                    name = "vlan/"+trunk
+                    result.append((name, trunk))
+                response[port] = result
+            native = self._interface_info(port.label)['tag']
+            if native != []:
+                response[port].append(("vlan/native", native))
+
+        return response
+
+    def revert_port(self, port):
+
+        shell_cmd_1 = 'sudo ovs-vsctl del-port {port}'.format(port=port)
+        shell_cmd_2 = 'sudo ovs-vsctl add-port {switch} {port}'.format(
+                    switch=self.hostname, port=port
+                    ) + ' vlan_mode=native-untagged'
+        self.ovs_connect(shell_cmd_1, shell_cmd_2)
+
+    def modify_port(self, port, channel, new_network):
+
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                return self._remove_native_vlan(interface)
+            else:
+                return self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if new_network is None:
+                return self._remove_vlan_from_port(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                return self._add_vlan_to_trunk(interface, vlan_id)
+
+# 2. Private Methods:
+
+    def _string_to_list(self, a_string):
+        """Converts a string representation of list to list.
+        Args:
+            a_string: list output recieved as string.
+                e.g. Strings starting with '[' and ending with ']'
+        Returns: object of list type.
+                 Empty list is put as None.
+        """
+        if a_string == '[]':
+            return ast.literal_eval(a_string)
+        else:
+            a_string = a_string.replace("[", "['").replace("]", "']")
+            a_string = a_string.replace(",", "','")
+            a_list = ast.literal_eval(a_string)
+            a_list = [ele.strip() for ele in a_list]
+            return a_list
```

Comment:
Could you put some examples of what `a_string` looks like? From the description, `sample = "['foo', 'bar']"` this should be converted succesfully, but this method fails to do so. This is what I get:
```
Traceback (most recent call last):
  File "test.py", line 23, in <module>
    print _string_to_list(sample)
  File "test.py", line 16, in _string_to_list
    a_list = ast.literal_eval(a_string)
  File "/usr/lib/python2.7/ast.py", line 49, in literal_eval
    node_or_string = parse(node_or_string, mode='eval')
  File "/usr/lib/python2.7/ast.py", line 37, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 1
    [''foo'',' 'bar'']
```
Simple calling `ast.literal_eval("['foo', 'bar']")` works.

Reply:
also, you could make these methods static since you aren't actually using the class instance here. but I won't block on that.

Reply:
I have explained in the previous comment why these functions (_string_to_list and _string_to_dict) belong inside the class. 
Also I have provided the docstring with the sample string it processes to convert it into list or dictionary respectively.  
Let me know what you and @zenhack  think about it.